userspace/libscap/scap.c

```diff
@@ -34,24 +36,24 @@ limitations under the License.
 #include <fcntl.h>
 #endif // _WIN32
 
-#ifdef CYGWING_AGENT
-#define DRAGENT_WIN_HAL_C_ONLY
-#include <dragent_win_hal_public.h>
-#endif
-
 #include "scap.h"
 #ifdef HAS_CAPTURE
-#ifndef CYGWING_AGENT
+#if !defined(_WIN32) && !defined(CYGWING_AGENT)
 #include "driver_config.h"
-#endif // CYGWING_AGENT
+#endif // _WIN32 && CYGWING_AGENT
 #endif // HAS_CAPTURE
 #include "../../driver/ppm_ringbuffer.h"
 #include "scap_savefile.h"
 #include "scap-int.h"
-#if defined(HAS_CAPTURE) && !defined(CYGWING_AGENT)
+#if defined(HAS_CAPTURE) && !defined(_WIN32) && !defined(CYGWING_AGENT)
 #include "scap_bpf.h"
 #endif
 
+#if defined(_WIN32) || defined(CYGWING_AGENT)
```

Comment:
This place is as good as any to complain about the complex `#ifdef`s again ;)

TBH, I'd remove CYGWING_AGENT support, it's completely bitrotten.

Also, how do _WIN32 and WDIG relate? Are they 1:1?

Reply:
CYGWING_AGENT: should I remove it here or let you do in your big refactory? It feels the refactory would be a better place to do this kind of cleanup.

_WIN32 and WDIG are two different things. The former is used in scap/sinsp to compile them under win32. The latter is an extension of UDIG on Windows and is used to include the fillers in the windows portal.